.github/workflows/ci.yml

```diff
@@ -66,73 +82,29 @@ jobs:
       run:
         shell: bash
     steps:
-      - name: Setup Container
+      - name: Setup container
         run: apt update && apt install -y git
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version: lts/*
-      - name: Install Dependencies
+      - name: Install project dependencies
         run: npm ci
-      - name: Run Unit Tests
+      - name: Run unit tests
         run: npm run test:coverage
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v3
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: code-coverage-report
+          path: coverage/
 
   jil-smoke-chrome:
```

Comment:
Should the tests still run on push to main?  If no, should we isolate the stuff that needs to happen on push to main to its own action?

Reply:
There is probably very little value in running these on a push to main.

Reply:
I can do that in another PR if we want that change. I just want to get the shared workflow in on this PR to test it out in github.